blocks/library/author/block.js

```diff
@@ -0,0 +1,108 @@
+*/
+import { connect } from 'react-redux';
+
+/**
+ * WordPress depensdencies
+ */
+import { Component, compose } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { PanelBody, withAPIData, Placeholder, Spinner } from '@wordpress/components';
+
+/**
+* Internal dependencies
+*/
+import InspectorControls from '../../inspector-controls';
+import ToggleControl from '../../inspector-controls/toggle-control';
+import BlockAlignmentToolbar from '../../block-alignment-toolbar';
+import BlockDescription from '../../block-description';
+import SingleAuthor from '../../single-author';
+import {
+	getCurrentPostAuthor,
+} from '../../../editor/selectors';
+
+class AuthorBlock extends Component {
+	constructor() {
+		super( ...arguments );
+		this.toggleHideBio = this.toggleHideBio.bind( this );
+		this.toggleHideAvatar = this.toggleHideAvatar.bind( this );
+		this.toggleHideName = this.toggleHideName.bind( this );
+	}
+	toggleHideBio() {
+		const { attributes, setAttributes } = this.props;
+		setAttributes( { hideBio: ! attributes.hideBio } );
+	}
+
+	toggleHideAvatar() {
+		const { attributes, setAttributes } = this.props;
+		setAttributes( { hideAvatar: ! attributes.hideAvatar } );
+	}
+
+	toggleHideName() {
+		const { attributes, setAttributes } = this.props;
+		setAttributes( { hideName: ! attributes.hideName } );
+	}
+
+	render() {
+		const { focus, attributes, setAttributes, author } = this.props;
+		const { align, hideBio, hideAvatar, hideName } = attributes;
+		return [
+			! author.data && (
+				<Placeholder key="placeholder"
+					icon="admin-post"
+					label={ __( 'Author' ) }
+				>
+					<Spinner />
+				</Placeholder>
+			),
+			author.data && (
+				<SingleAuthor key="content"
+					avatar={ hideAvatar ? null : author.data.avatar_urls[ '96' ] }
+					bio={ hideBio ? null : author.data.description }
+					className={ align && `align${ align }` }
+					name={ hideName ? null : author.data.name }
+				/>
+			),
+			focus && (
+				<InspectorControls key="inspector">
+					<BlockDescription>
+						<p>{ __( 'Shows the post author' ) }</p>
+					</BlockDescription>
+					<PanelBody title={ __( 'Alignment' ) }>
+						<BlockAlignmentToolbar
+							value={ align }
+							onChange={ ( nextAlign ) => setAttributes( { align: nextAlign } ) }
+						/>
+					</PanelBody>
+					<ToggleControl
+						label={ __( 'Hide bio' ) }
+						checked={ hideBio }
+						onChange={ this.toggleHideBio }
+					/>
+					<ToggleControl
+						label={ __( 'Hide avatar' ) }
+						checked={ hideAvatar }
+						onChange={ this.toggleHideAvatar }
+					/>
+					<ToggleControl
+						label={ __( 'Hide name' ) }
+						checked={ hideName }
+						onChange={ this.toggleHideName }
+					/>
+				</InspectorControls>
+			),
+		];
+	}
+}
+
+export default compose(
+	connect(
+		( state ) => ( {
+			postAuthorId: getCurrentPostAuthor( state ),
+		} )
+	),
+	withAPIData( ( props ) => ( {
+		author: `/wp/v2/users/${ props.postAuthorId }`,
```

Comment:
We're going to have permissions issues here for non-administrator users:

https://github.com/WordPress/wordpress-develop/blob/53f1c74e1c55910f9b7e4aebc8033571d3e52df8/src/wp-includes/rest-api/endpoints/class-wp-rest-users-controller.php#L357-L384

I'm not sure how we'd address this.

Reply:
This is a complex problem, that probably will be faced in other blocks.
I'm seeing two possible paths:
We conclude that this information should be available in the API and we create a patch for that. To me author name, description and avatar look like information that can be public.
We conclude that this information should not be public. In this case, we need to have a mechanism to only allow the author of the post to add this block. And we need a state that shows some message like "No permissions to display this block in the editor, please use the preview to visualize this block content" when the post is being edited by a user that is not admin or post author.